### PR TITLE
vehicle_attitude only based yaw control fix

### PR DESCRIPTION
**Pull request by jgoppert**
@Stifael this is a follow up to #10683. I've tested this in sitl and now have working yaw control again with vehicle attitude and w/o vehicle_local_position. Thanks for the work on your end.

### CODE REVIEW

**Comment by TSC21** [comment]
@jgoppert can you please rebase?

**Comment by jgoppert** [comment]
@TSC21 It was already rebased on master. I tried to restart the failed tests. It seems some of the jenkins boxs are out of space?

**Comment by TSC21** [comment]
Yeah it seems so. @dagar?

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 1]
```diff
@@ -476,68 +484,68 @@ MulticopterPositionControl::limit_altitude(vehicle_local_position_setpoint_s &se
 void
 MulticopterPositionControl::set_vehicle_states(const float &vel_sp_z)
 {
-	if (_local_pos.timestamp == 0) {
-		return;
+	if (_att.timestamp != 0) {
+		if (PX4_ISFINITE(_att.q[0])) {
```
[the remark this answers is not in the document]
Comment (jgoppert) [reply src/modules/mc_pos_control/mc_pos_control_main.cpp]: I think this follows the existing convention. If the attitude quaternion is Nan the yaw state won't update. See line 539 below where it used to do the same with local_pos .

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 2]
```diff
@@ -433,6 +435,12 @@ MulticopterPositionControl::poll_subscriptions()
 		orb_copy(ORB_ID(vehicle_local_position), _local_pos_sub, &_local_pos);
 	}
 
+	orb_check(_att_sub, &updated);
+
+	if (updated) {
+		orb_copy(ORB_ID(vehicle_attitude), _att_sub, &_att);
```
Comment (dagar) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]:
```suggestion
		vehicle_attitude_s att;
		if (orb_copy(ORB_ID(vehicle_attitude), _att_sub, &att) == PX4_OK) {
			if (PX4_ISFINITE(_att.q[0])) {
				_states.yaw = Eulerf(Quatf(_att.q)).psi();
			}
		}
```

File: src/modules/mc_pos_control/mc_pos_control_main.cpp [hunk 3]
```diff
@@ -550,6 +558,7 @@ MulticopterPositionControl::run()
 	_control_mode_sub = orb_subscribe(ORB_ID(vehicle_control_mode));
 	_params_sub = orb_subscribe(ORB_ID(parameter_update));
 	_local_pos_sub = orb_subscribe(ORB_ID(vehicle_local_position));
+	_att_sub = orb_subscribe(ORB_ID(vehicle_attitude));
```
Comment (dagar) [inline src/modules/mc_pos_control/mc_pos_control_main.cpp]: Missing orb_unsubscribe. Line ~ 810 below.

**Comment by jgoppert** [comment]
@dagar Addressed your review and streamlined the changes.

**Comment by dagar** [comment]
Thanks, one last minor thing that just occurred to me is to initialize _states. Then we can't even potentially end up with garbage in yaw in the unlikely event of a delayed vehicle_attitude for whatever reason.

File: src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp [hunk 4]
```diff
@@ -90,6 +90,11 @@ void FlightTask::_evaluateVehicleLocalPosition()
 	_yaw = NAN;
 	_dist_to_bottom = NAN;
 
+	if ((_time_stamp_current - _sub_attitude->get().timestamp) < _timeout) {
+		// yaw
+		_yaw = matrix::Eulerf(matrix::Quatf(_sub_attitude->get().q)).psi();
```
Comment (bkueng) [inline src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp]: Can you remove the yaw assignment further down? https://github.com/PX4/Firmware/pull/10803/files#diff-76c7cef38691d384b72e5f1f8695eb69R122

Comment (jgoppert) [reply src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp]: Yes, I believe so, let's see if the tests  pass.

Comment (jgoppert) [reply src/lib/FlightTasks/tasks/FlightTask/FlightTask.cpp]: It flies fine in SITL using ekf2.

**Pull request merged**